dice-game-ui-angular/src/app/saved-games/saved-games.component.html

```diff
@@ -0,0 +1,24 @@
+<h1 align="center" class="text-muted">Saved games</h1>
+
+<div *ngIf="games.length>0; else noSavedGames">
+    <mat-card class="m-1" *ngFor="let ge of games">
+        <mat-card-header>
+            <mat-card-title>
+                <button mat-flat-button [routerLink]="['/game', ge.guid]">
```

Comment:
It is better to make a function to get a link from game object.

Reply:
I'm not completely understand what you mean. ((

Reply:
Make a function in a component code that will return a link for the router which will navigate to a game with such guid. Like
```ts
getGameLink(gameEngine: GameEngine) {
  return `/game/${gameEngine.guid}`; // Or return ['/game', gameEngine.guid];
}
```

Reply:
Honestly, It's a really strange and unnatural decision for me. I can stand at least for creating a method which derectly navigate to the page (as a part shifting control to the component) but your decision doesn't look as a good practice for me. Of course, you have more experience here and if you say that would be better, I'll change it.

Reply:
Common practice is to not have much logic in a template, because it means mixing of presentation and behavior which is a violation of some good programming principles. Your template doesn't need to know how exactly you calculate a game link, it just need to know that it exists.

Reply:
For example, in some large projects you can have a single service that manages all navigation links and build needed link with parameters if needed. If you decide to do this in your project, you will just need to change a component method without changing a template.

Reply:
Ok. I changed.